### DISCUSSION TITLE
Run a system before the bevy application exits

### DISCUSSION THREAD

**Discussion by Akaam-Zain** [bevyengine/bevy#5281] [Q&A] [answered]
I have client-server bevy application which communicates through a TCP connection. When the client window is closed by the client - before the application actually exits I want to send a message to the server indication that the client is terminated. For that I need to know how I could add a system just before the bevy application exits.

Appreciate your help. Thanks!

**Comment by tim-blackbird** [comment, marked as answer]
You can listen for an [AppExit](https://docs.rs/bevy/latest/bevy/app/struct.AppExit.html) event in a system in the `PostUpdate` stage.

**Reply by Akaam-Zain** [reply]
Hey, appreciate your answer. But would be grateful if you could elaborate on how it could be done. I am quite new to Bevy.
Here's my current implementation 

```
fn exit_system(mut connection: ResMut<Connection>) {

    let events = Events::<AppExit>::default();
    let mut reader = events.get_reader();

    if reader.iter(&events).count() > 0 {
        //... my implmentation
    }
```

And I call the above system as:
 `.add_system_to_stage(CoreStage::PostUpdate, exit_system)`

**Reply by NiklasEi** [reply]
You should get the `EventReader` as a parameter of your system.
See for example https://github.com/bevyengine/bevy/blob/a1d3f1b3b42c3a6e05a532362ddfccf0be6b5df3/examples/app/drag_and_drop.rs#L12

**Reply by Akaam-Zain** [reply]
@NiklasEi and @devil-ira thank you - much appreciated. That solved the problem.
